### ISSUE TITLE
"View Source" button for @property, @cached_property methods

### ISSUE THREAD

**Issue by tmeyier** [mitmproxy/pdoc#653]
#### Problem Description
pdoc nicely collects docstrings from both @property as well as @cached_property methods and shows them as documentation for the respective instance attributes. But it doesn't give a "View Source" button to view the source code of the corresponding methods.
I'm not sure if this is intended behaviour. I also couldn't find any directly related issue. Anyway, I'd like to suggest to add a "View Source" button in both cases (and maybe for other related decorators?). I'd say that usually an attribute is a @property (instead of just an instance variable) because there is something interesting inside the @property method, that a reader of the API documentation may well want to see.
I've looked into the source code of pdoc and found the option to produce "View Source" buttons used only in the jija template. But I couldn't figure out:
* do I just need to add similar code somewhere else in the jinja template (where attributes are treated)?
* or is extra handling needed to extract the source code of a @property in the parsing stage of pdoc and associate it to the corresponding attribute? 
Maybe, you could give me a hint, then I'd try and provide a pull request.

#### Steps to reproduce the behavior:

Save this to bla.py:

```python
from functools import cached_property


class A:
    """An example class to be documented by pdoc."""

    @property
    def la():
        """Print la."""
        print("la")

    @cached_property
    def li():
        """Print li."""
        print("li")
```

then:

```bash
pdoc bla.py
```

#### System Information
Paste the output of "pdoc --version" here.

**Comment by mhils** [comment]
You'd need to make this method work for variables: https://github.com/mitmproxy/pdoc/blob/28d394e557edf7306df404f6fa70b1ea19ac3ad9/pdoc/doc.py#L162-L170

This is a bit of a non-trivial undertaking unfortunately because `inspect.getsource` only works with classes and functions AFAIK.

**Comment by tmeyier** [comment]
I have created a branch here:
https://github.com/tmeyier/pdoc/tree/show_property_source 

I adapted the code to add the property function source to the corresponding Variable object.

With that, I seem to get the behavior I was looking for. Also, the adapted pdoc is still able to document itself (and shows source code of properties in there, too).

Could you have a look, please?

(I've uploaded to github for the first time. Is this the intended way how to do it? Also: do you even have access to the fork+branch I produced?)

**Comment by mhils** [comment]
Awesome! The best way to contribute these changes is to open a pull request, which I strongly encourage you to do! :)

**Comment by tmeyier** [comment]
See #654

**Comment by tmeyier** [comment]
What to do in case of a read-write property? It's probably possible to extract the source code of both the getter and the setter function. Should they be concatenated in the generated html?

**Comment by mhils** [comment]
Awesome, thanks!

> What to do in case of a read-write property or a property with a deleter function?

I think that's a separate concern from the "view source" thing. I think the current behavior (picking up the last docstring) is sensible enough to be honest. My hot take on this is that you shouldn't be using `@property` if the logic in your getters/setters/deleters is so complex that you need to describe them separately. I realize that belief is not shared by everyone, but concatenated docstrings will always end up quite ugly I think. :)

**Comment by tmeyier** [comment]
About the sources of setter and deleter of a @property: 

I tried a quick hack concatenating the three sources with "\n" in between. The result looks reasonable. But then I'd have to do something with the line numbers and it's getting complicated. 
Still, from an API documentation point of view, I think it would be good to be able to distinguish read-only properties from properties, that you can assign to. But that's definitely another issue.

**Comment by mhils** [comment]
Shipped as pdoc 14.3! 😃
